jsonrpc/server.go

```diff
@@ -4,13 +4,16 @@ package jsonrpc
 import (
 	"bufio"
 	"bytes"
+	"context"
 	"encoding/json"
 	"errors"
 	"io"
 	"reflect"
 	"strings"
+	"sync"
 
 	"github.com/go-playground/validator/v10"
+	"golang.org/x/sync/errgroup"
```

Comment:
Pros:
- allows to receive errors from goroutines
Cons:
- there is no way to limit number of spawned goroutines out of the box

Reply:
how about https://pkg.go.dev/github.com/sourcegraph/conc/pool#ErrorPool, we use this package already

Reply:
nice, I'll use this one